server/handlers/order.js

```diff
@@ -0,0 +1,63 @@
+var dao = require("../dao/dao");
+const { Order } = require("../models/order");
+const { validationResult } = require("express-validator");
+const {
+  clientIDPathValidator,
+  productIDPathValidator,
+  productQtyPathValidator,
+  productsValidator,
+} = require("./shared_validators");
+
+exports.createOrderValidatorChain = [
+  clientIDPathValidator,
+  productsValidator,
+  productIDPathValidator,
+  productQtyPathValidator,
+];
+
+exports.createOrderHandler = async function (req, res, next) {
+  // Insert the new order
+
+  // productPrice is hardcoded to 1 as a tempoarary solution for now, will be fixed in the next sprints
+
+  var result;
+  var totalPrice = 0;
+  var productPrice = 1;
+
+  var isNegativeQuantity = false;
+  req.body.products.forEach((product) => {
+    totalPrice = totalPrice + product.quantity * productPrice;
+    if (product.quantity < 0) {
+      isNegativeQuantity = true;
+    }
+  });
+
+  if (isNegativeQuantity) {
```

Comment:
remove this code, for the reason above. A negative quantity is an argument error that must be handled in the validation phase.